### PR TITLE
Intercept the "edit code" command from WebView and run the native JetBrains UI

### PR DESCRIPTION
**Pull request by dominiccooney**
Out of the box, the WebView runs the Cody "edit" command which tries to open a QuickPick. VSCode shims does not support QuickPick, and has an independent implementation of the edits UI that is not integrated at the extension TypeScript level.

For now, intercept the WebView's request to run "edit" and invoke the native JetBrains edit command directly.

Fixes CODY-3238.

## Test plan

Manually tested:

1. Open a file.
2. Open the sidebar, commands tab, click Edit Code.
3. Verify that the Cody Edit dialog appears.

Note, the OK button doesn't work because of an unrelated regression in JetBrains edits in general, probably introduced in https://github.com/sourcegraph/cody/commit/698239e099a153d54364960e89e6e5402843e820 . When that is fixed, this will work end-to-end.

### CODE REVIEW

File: src/main/kotlin/com/sourcegraph/cody/ui/CodyToolWindowFactory.kt [hunk 1]
```diff
@@ -275,6 +280,27 @@ class WebUIHostImpl(
               .showSettingsDialog(project, AccountConfigurable::class.java)
         }
       }
+      // TODO: Delete this intercept when Cody edits UI is abstracted so JetBrains' native UI can be invoked from the
+      // extension TypeScript side through Agent.
+      "{\"command\":\"command\",\"id\":\"cody.action.command\",\"arg\":\"edit\"}" -> {
```
Comment (pkukielka) [inline src/main/kotlin/com/sourcegraph/cody/ui/CodyToolWindowFactory.kt]: This looks a bit fragile, as all parsing jsons by hand.
Are there any downsides to parse it like I did in `handleCefQuery`?

**Review by pkukielka — APPROVED** [review]
Giving you an approve if you want to merge, but I would consider proper JSON parsing - it won't be any harder but is potentially safer.

Also, you need to reformat with spotless ;)

File: src/main/kotlin/com/sourcegraph/cody/ui/CodyToolWindowFactory.kt (hunk 1 above)
Comment (dominiccooney) [reply src/main/kotlin/com/sourcegraph/cody/ui/CodyToolWindowFactory.kt]: The serialization comes from JSON.stringify so this is not particularly fragile, and this is succinct.

**Comment by dominiccooney** [comment]
- Run spotless. Spotless comment line breaking is odd.
- Switch to JSON parsing, did not bother to check whether any new keys have appeared because this is already getting verbose.

**Pull request merged**